tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7779,10 +7779,10 @@ def err_hlsl_unsupported_keyword_for_min_precision : Error<
    "%0 is only supported with -enable-16bit-types option">;
 def warn_hlsl_derivatives_in_wrong_shader_kind: Warning<
    "Intrinsic %0 potentially used by %1 requires derivatives - only available in pixel, compute, amplification, mesh, or broadcast node shaders.">,
-    DefaultError, InGroup<HLSLAvailability>;
+    InGroup<HLSLAvailability>;
 def warn_hlsl_derivatives_in_wrong_shader_model: Warning<
    "Intrinsic %0 potentially used by %1 requires derivatives - shader model 6.6 or greater is required for use in compute, amplification, or mesh shaders.">,
-    DefaultError, InGroup<HLSLAvailability>;
+    InGroup<HLSLAvailability>;
 def warn_hlsl_derivatives_wrong_numthreads : Warning<
```

Comment:
This one and the next are in the same group and are still `DefaultError`.  This one could still impact existing pre-SM 6.8 usage.

The next one is `warn_hlsl_intrinsic_overload_in_wrong_shader_model` which is the only one that (for now) seems to be limited to diagnosing the SM 6.8 overload, and is the case where I think you wanted DefaultError.  But the warning itself is generic enough that you could certainly see it being applied to other overload cases that have different shader model requirements.

It feels weird to have warnings in the same group with some DefaultError and others not, when the granularity you have to change the default is per-group.

What's the consistent policy here?  We haven't applied this to SM 6.8 ops and overloads generally, nor are we limiting it to when you compile to SM 6.8, nor is it triggered by a language version update.  It's not easy to explain the policy to users now.  I'm not even sure what it's supposed to be now.

Is it this?
> Some new SM 6.8 operation overloads may be diagnosed early.  This may cause a compilation failure even if the operation would not have been used in the final shader after optimizations and dead code elimination.
_Followed by mitigation instructions..._